src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -139,14 +139,40 @@ public static void RegisterInstance(VisualStudioInstance instance)
         /// </param>
         public static void RegisterMSBuildPath(string msbuildPath)
         {
-            if (string.IsNullOrWhiteSpace(msbuildPath))
+            RegisterMSBuildPath(new string[] {
+                msbuildPath
+#if NET46
+                // Finds and loads NuGet assemblies if msbuildPath is in a VS installation
+                , Path.GetFullPath(Path.Combine(msbuildPath, "..", "..", "..", "Common7", "IDE", "CommonExtensions", "Microsoft", "NuGet"))
+#endif
+            });
+        }
+
+        /// <summary>
+        ///     Add assembly resolution for Microsoft.Build core dlls in the current AppDomain from the specified
+        ///     path.
+        /// </summary>
+        /// <param name="msbuildSearchPaths">
+        ///     Paths to directories containing a deployment of MSBuild binaries.
+        ///     A minimal MSBuild deployment would be the publish result of the Microsoft.Build.Runtime package.
+        ///
+        ///     In order to restore and build real projects, one needs a deployment that contains the rest of the toolchain (nuget, compilers, etc.).
+        ///     Such deployments can be found in installations such as Visual Studio or dotnet CLI.
+        /// </param>
+        public static void RegisterMSBuildPath(string[] msbuildSearchPaths)
+        {
+            if (msbuildSearchPaths.Length < 1)
+            {
+                throw new ArgumentException("Must provide at least one search path to RegisterMSBuildPath.");
+            }
+            if (msbuildSearchPaths.Any(string.IsNullOrWhiteSpace))
             {
-                throw new ArgumentException("Value may not be null or whitespace", nameof(msbuildPath));
+                throw new ArgumentException("Value may not be null or whitespace", nameof(msbuildSearchPaths));
```

Comment:
Include index of bogus entry?

Reply:
0-indexed or 1-indexed? I made it 1-indexed but would be happy to change it.